src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -1088,3 +1088,17 @@ def update_local_user(cmd, client, resource_group_name, account_name, username,
                          home_directory, has_shared_key, has_ssh_key, has_ssh_password)
     return client.create_or_update(resource_group_name=resource_group_name, account_name=account_name,
                                    username=username, properties=local_user)
+
+
+def begin_failover(client, resource_group_name, account_name, failover_type=None, yes=None, **kwargs):
+    if not yes and failover_type and failover_type.lower() != "planned":
+        from knack.prompting import prompt_y_n
+        message = """
+        The secondary cluster will become the primary cluster after failover. Please understand the following impact to your storage account before you initiate the failover:
+            1. Please check the Last Sync Time using `az storage account show` with `--expand geoReplicationStats` and check the "geoReplicationStats" property. This is the data you may lose if you initiate the failover.
+            2. After the failover, your storage account type will be converted to locally redundant storage (LRS). You can convert your account to use geo-redundant storage (GRS).
+            3. Once you re-enable GRS/GZRS for your storage account, Microsoft will replicate data to your new secondary region. Replication time is dependent on the amount of data to replicate. Please note that there are bandwidth charges for the bootstrap. Please refer to doc: https://azure.microsoft.com/pricing/details/bandwidth/
+        """
+        if not prompt_y_n(message):
+            return None
```

Comment:
Current behavior for prompt `n` is throwing an exception saying `Operation Cacelled` instead of returning nothing. The command will exit with error status.

Additionally, better to utilize `user_confirmation` defined in cli core to handle in case no tty available:
https://github.com/Azure/azure-cli/blob/085daccc19445387586e4f532b325f51e526eaff/src/azure-cli-core/azure/cli/core/util.py#L1196-L1205